app/controllers/api/proxy.php

```diff
@@ -55,14 +55,25 @@
     ->inject('dbForPlatform')
     ->inject('dbForProject')
     ->action(function (string $domain, string $resourceType, string $resourceId, Response $response, Document $project, Certificate $queueForCertificates, Event $queueForEvents, Database $dbForPlatform, Database $dbForProject) {
+
         $mainDomain = System::getEnv('_APP_DOMAIN', '');
         if ($domain === $mainDomain) {
             throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'You cannot assign your main domain to specific resource. Please use subdomain or a different domain.');
         }
 
-        $functionsDomain = System::getEnv('_APP_DOMAIN_FUNCTIONS', '');
-        if ($functionsDomain != '' && str_ends_with($domain, $functionsDomain)) {
-            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'You cannot assign your functions domain or it\'s subdomain to specific resource. Please use different domain.');
+        $functionsDomain = System::getEnv('_APP_DOMAIN_FUNCTIONS');
+        $denyListString = System::getEnv('_APP_CUSTOM_DOMAIN_DENY_LIST');
+
+        if (!empty($functionsDomain)) {
+            $denyListString .= ',' . $functionsDomain;
+        }
```

Comment:
it should be the other way. 

if !empty(denyListString) {
     functionsDomain + denyListString
}